src/main/kotlin/blackjack/application/BlackJack.kt

```diff
@@ -1,43 +1,103 @@
 package blackjack.application
 
-import blackjack.application.dto.PlayerResult
-import blackjack.application.dto.PlayerResults
-import blackjack.application.dto.PlayerStatus
-import blackjack.application.dto.PlayerStatuses
+import blackjack.application.dto.BlackJackScore
+import blackjack.application.dto.BlackJackStatus
+import blackjack.application.dto.BlackJackStatuses
+import blackjack.application.dto.BlackJackWinningResult
+import blackjack.application.dto.BlackJackWinningResults
+import blackjack.application.dto.BlackjackScores
 import blackjack.domain.card.CardDeck
 import blackjack.domain.card.setupCardDeck
-import blackjack.domain.player.Player
-import blackjack.domain.player.Players
+import blackjack.domain.participant.Dealer
+import blackjack.domain.participant.Player
+import blackjack.domain.participant.Players
+import blackjack.domain.participant.vo.Name
 
-private fun Players.getPlayerStatuses(): PlayerStatuses =
-    players.map { PlayerStatus(it.name, it.cardsInHand.cards) }.let(::PlayerStatuses)
+private fun Dealer.status(): BlackJackStatus = BlackJackStatus(this.name, listOf(this.cardsInHand.cards.first()))
 
-private fun Players.getPlayerResults(): PlayerResults =
-    players.map { PlayerResult(it.name, it.cardsInHand.cards, it.cardsInHand.calculateScore()) }.let(::PlayerResults)
+private fun Players.statuses(): List<BlackJackStatus> =
+    players.map { BlackJackStatus(it.name, it.cardsInHand.cards) }
+
+private fun Dealer.score(): BlackJackScore =
+    BlackJackScore(this.name, this.cardsInHand.cards, this.cardsInHand.calculateScore())
+
+private fun Players.scores(): List<BlackJackScore> {
+    return players.map { BlackJackScore(it.name, it.cardsInHand.cards, it.cardsInHand.calculateScore()) }
+}
+
+private fun Dealer.winningResult(): BlackJackWinningResult =
+    BlackJackWinningResult(this.name, this.winningScores)
+
+private fun Players.winningResults(): List<BlackJackWinningResult> =
+    players.map { BlackJackWinningResult(it.name, it.winningScores) }
 
 class BlackJack private constructor(
+    private val dealer: Dealer,
     private val players: Players,
     private val cardDeck: CardDeck
 ) {
-    val statuses: PlayerStatuses
-        get() = players.getPlayerStatuses()
+    val statuses: BlackJackStatuses
+        get() = BlackJackStatuses(
+            buildList {
+                add(dealer.status())
+                addAll(players.statuses())
+            }
+        )
```

Comment:
코틀린 연산자를 이용해서 아래처럼 줄여볼 수 있겠네요 :)
```suggestion
        get() = BlackJackStatuses(players.statuses() + dealer.status())
```

Reply:
list 에서도 + operator 를 사용할수 있네요 😄